src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java

```diff
@@ -277,28 +288,41 @@ public boolean isBlockPowered(int x, int y, int z) {
 
         Entity entity = null;
 
-        try {
-            entity = ConstructorUtils.invokeConstructor(type.getEntityClass(), this);
-            entity.setLocation(entity.getLocation().setPosition(position));
-        } catch (Exception e) {
-            Sponge.getLogger().error(ExceptionUtils.getStackTrace(e));
+        Class<? extends Entity> entityClass = type.getEntityClass();
+
+        if (entityClass.isAssignableFrom(EntityPlayerMP.class) || entityClass.isAssignableFrom(EntityDragonPart.class)) {
+            // Unable to construct these
+            return Optional.absent();
+        }
+
+        net.minecraft.world.World world = (net.minecraft.world.World) (Object) this;
+
+        // Not all entities have a single World parameter as their constructor
+        if (entityClass.isAssignableFrom(EntityLightningBolt.class)) {
+            entity = (Entity) new EntityLightningBolt(world, position.getX(), position.getY(), position.getZ());
+        } else if (entityClass.isAssignableFrom(EntityEnderPearl.class)) {
+            EntityArmorStand tempEntity = new EntityArmorStand(world, position.getX(), position.getY(), position.getZ());
```

Comment:
I meant to unset this after the constructor and have an `UnknownProjectileSource` as the default. I'll fix this now.